packages/trader/src/Modules/SmartChart/Components/all-markers.jsx

```diff
@@ -141,9 +143,11 @@ const shadowed_text = ({ ctx, is_dark_theme, text, left, top, scale }) => {
     ctx.textAlign = 'center';
     const size = Math.floor(scale * 12);
     ctx.font = `bold ${size}px BinarySymbols, Roboto`;
-    ctx.shadowColor = is_dark_theme ? 'rgba(16,19,31,1)' : 'rgba(255,255,255,1)';
-    ctx.shadowBlur = 12;
-    for (let i = 0; i < 5; ++i) {
+    if (!is_firefox) {
+        ctx.shadowColor = is_dark_theme ? 'rgba(16,19,31,1)' : 'rgba(255,255,255,1)';
+        ctx.shadowBlur = 12;
+    }
+    for (let i = 0; i < (is_firefox ? 1 : 5); ++i) {
```

Comment:
5? 1? replace these magical numbers with named constants?

Reply:
`5` was the default amount of times we want the loop to run for `fillText` (as per original UI for the canvas text with shadow blur), due to Firefox suffering from this, we disabled the shadowBlur, so we only need to do `fillText` once. As suggested by @aminroosta 's review.

Didn't make it a`const` since it's not supposed to be a permanent solution, ideally, we shouldn't need the `is_firefox` hack anymore if we can come up with a more performant method (for Firefox) of rendering text in `canvas`.

I'll just add a comment above the code line to explain the reasons why this is needed.

Reply:
But why 5 times?

Reply:
From my understanding of `canvas` text rendering, if u add blurred shadows to text it isn't visible enough until u layer it multiple times (in this case he's adding 4 extra layers) to increase the shadowBlur intensity.

Feel free to correct me if I'm wrong @aminroosta .

Reply:
ok makes sense 👍